Fluid/TemplateContext.cs

```diff
@@ -130,5 +130,10 @@ public void SetValue(string name, object value)
         {
             SetValue(name, FluidValue.Create(value));
         }
+
+        public void SetValue<T>(string name, Func<T> factory)
+        {
+            SetValue(name, new FactoryValue<T>(factory));
```

Comment:
Should we provide an overload that passes in the `TemplateContext`?

Reply:
Not possible